### PR TITLE
Add component for night vision sprite visibility

### PR DESCRIPTION
**Pull request by ShadowCommander**
## About the PR

Moves which entities are visible in night vision to its own component.

This changes it from all MobStateComponent entities to RMCNightVisionVisibleComponent.

Added RMCNightVisionVisibleComponent to CMBaseMobSpeciesOrganic, RMCSmallHost, CMXenoBase, and XenoNest prototypes.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by ShadowCommander** [comment]
Having some trouble with sprites on the same tile rendering in the wrong order.

File: Content.Client/_RMC14/NightVision/NightVisionOverlay.cs [hunk 1]
```diff
@@ -41,23 +39,37 @@ protected override void Draw(in OverlayDrawArgs args)
         var eye = args.Viewport.Eye;
         var eyeRot = eye?.Rotation ?? default;
 
-        // TODO RMC14 this should use its own component
-        var entities = _entity.EntityQueryEnumerator<MobStateComponent, SpriteComponent, TransformComponent>();
-        while (entities.MoveNext(out var uid, out _, out var sprite, out var xform))
+        var entries = new List<NightVisionRenderEntry>();
```
Comment (DrSmugleaf) [inline Content.Client/_RMC14/NightVision/NightVisionOverlay.cs:42]: This should be turned into a readonly field that gets cleared at the start of Draw, for less allocations

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Content.Client/_RMC14/NightVision/NightVisionOverlay.cs (hunk 1 above)
Comment (ShadowCommander) [reply Content.Client/_RMC14/NightVision/NightVisionOverlay.cs]: Done

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**